infrastructure/src/main/java/org/corfudb/infrastructure/log/statetransfer/StateTransferManager.java

```diff
@@ -0,0 +1,293 @@
+
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.Lists;
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.Builder;
+import lombok.Builder.Default;
+import lombok.EqualsAndHashCode;
+import lombok.EqualsAndHashCode.Exclude;
+import lombok.Getter;
+import lombok.NonNull;
+import lombok.ToString;
+import lombok.extern.slf4j.Slf4j;
+import org.corfudb.infrastructure.log.StreamLog;
+import org.corfudb.infrastructure.log.statetransfer.batch.TransferBatchRequest;
+import org.corfudb.infrastructure.log.statetransfer.batch.TransferBatchResponse;
+import org.corfudb.infrastructure.log.statetransfer.batch.TransferBatchResponse.TransferStatus;
+import org.corfudb.infrastructure.log.statetransfer.batchprocessor.StateTransferBatchProcessor;
+import org.corfudb.infrastructure.log.statetransfer.exceptions.TransferSegmentException;
+
+import java.util.Iterator;
+import java.util.List;
+import java.util.Optional;
+import java.util.Set;
+import java.util.stream.LongStream;
+import java.util.stream.Stream;
+
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.FAILED;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.NOT_TRANSFERRED;
+import static org.corfudb.infrastructure.log.statetransfer.StateTransferManager.TransferSegmentStatus.SegmentState.TRANSFERRED;
+
+/**
+ * A class responsible for managing a state transfer on the current node.
+ * It executes the state transfer for each non-transferred segment synchronously.
+ */
+@Slf4j
+@Builder
+public class StateTransferManager {
+
+    /**
+     * A data class that represents a non-empty and bounded segment to be transferred.
+     */
+    @EqualsAndHashCode
+    @Getter
+    @ToString
+    @Builder
+    @AllArgsConstructor(access = AccessLevel.PRIVATE)
+    public static class TransferSegment implements Comparable<TransferSegment> {
+        /**
+         * Start address of a segment range to transfer, inclusive and non-negative.
+         */
+        private final long startAddress;
+        /**
+         * End address of a segment range to transfer, inclusive and non-negative.
+         */
+        private final long endAddress;
+        /**
+         * A status of a transfer of a segment.
+         */
+        private final TransferSegmentStatus status;
+
+        @Override
+        public int compareTo(TransferSegment other) {
+            return Long.compare(this.getStartAddress(), other.getStartAddress());
+        }
+
+        /**
+         * Compute the total number of transferred addresses.
+         * {@link #endAddress} and {@link #startAddress} can only be non-negative longs such that
+         * {@link #endAddress} >= {@link #startAddress}.
+         *
+         * @return Sum of the total addresses transferred.
+         */
+        public long computeTotalTransferred() {
```

Comment:
The name is a little misleading, since the value denotes progress, but its actually a non-changing value that doesn't reflect progress.

Reply:
Fixed